### DISCUSSION TITLE
log4net and SCOM

### DISCUSSION THREAD

**Discussion by Jjmboni** [apache/logging-log4net#173] [Q&A] [answered]
Log4Net v2.0.16 does not play well with SCOM, to the point where if you copy log4net.dll to an IIS Web Server containing SCOM it will crash w3wp.exe.  Doesn't even have to be used.

I've been forced to downgrade to v2.0.8 which works but has its own known vulnerabilities.

My question is: Does v2.0.17 work with SCOM?  Or will v3.0.0 you're planning on releasing in September work with SCOM?

EDIT: I was incorrect saying it was SCOM.  The application causing conflicts is Microsoft Monitoring Agent - "perfmon.dll".  It may be installed via SCOM, I don't know.

**Comment by FreeAndNil** [comment]
Hi @Jjmboni,

I assume you mean Microsoft System Center Operation Manager with SCOM?

We have no experience with this product and no test system for reproducing your issue.

You can try the latest Preview and see if the problem persists: https://github.com/apache/logging-log4net/releases/tag/rc%2F3.0.0-preview.2
In case the problem is still there (which I asume), you can try to find out which exact version introduces the problem.
Then we can try to guess which change is relevant.

Do you have crash dumps or event log entries?

**Reply by Jjmboni** [reply]
Happy to help!  I'll give you as much information as I can.

But first, my apologies; I was a little incorrect in my original post.  The conflict in question is Microsoft Monitoring Agent "perfmon.dll".  It may be installed by SCOM, I'm not sure.  I'm working a little hampered by the fact that I'm a software developer and not privy to what apps the Server Team is installing.  But I have admin access to my server so I can find out details.

I'm basing this on the following two posts:

https://stackoverflow.com/questions/74878576/microsoft-monitoring-agents-perfmon-dll-crashing-iis-web-application-pool-while
https://learn.microsoft.com/en-us/archive/msdn-technet-forums/17f589b0-97be-414e-84f7-40703a0c444f

...as well as my own experience with log4net and MMA.

The version of MMA installed on my server is v10.19.10014.0.  So looks like even though those two posts are older I'm still getting errors with the latest version.

The specific error thrown in the Event Log is the same as shown in the microsoft.com url above.  I'll post mine which is very similar:

Faulting application name: w3wp.exe, version: 10.0.20348.1, time stamp: 0x405e4c14
Faulting module name: PerfMon64.dll, version: 10.19.10014.0, time stamp: 0x5c45e2de
Exception code: 0xc0000409
Fault offset: 0x000000000013f68f
Faulting process id: 0x2f40
Faulting application start time: 0x01dafa5c14ec0d2f
Faulting application path: c:\windows\system32\inetsrv\w3wp.exe
Faulting module path: C:\Program Files\Microsoft Monitoring Agent\Agent\APMDOTNETAgent\V10.19.10014.0\PerfMon64.dll
Report Id: d8eb245c-c049-4bd6-a44a-43563c803ebb
Faulting package full name: 
Faulting package-relative application ID: 

I have deployed all the versions of log4net from v2.0.8 to v2.0.17 and found the following:

v2.0.8 / 3/8/2017: No error.  Perfmon and MMA plays nicely.
v2.0.9 / 7/13/2020: Perfmon error, crashes w3wp.exe
v2.0.10 / 9/6/2020: Perfmon error, crashes w3wp.exe 
v2.0.11 / 9/18/2020: Perfmon error, crashes w3wp.exe
v2.0.12 / 10/18/2020: Perfmon error, crashes w3wp.exe
v2.0.13 / 10/28/2021: Perfmon error, crashes w3wp.exe
v2.0.14 / 12/17/2021: Perfmon error, crashes w3wp.exe
v2.0.15 / 7/25/2022: Perfmon error, crashes w3wp.exe
v2.0.16 / 3/1/2024: Perfmon error, crashes w3wp.exe
v2.0.17 / 3/15/2024: Perfmon error, crashes w3wp.exe
And finally: Downgrade back to v2.0.8 (to prove it's not something else): Website back to functioning normally, no more errors.

**Comment by FreeAndNil** [comment]
Which framework version of the log4net assembly do you use?
![image](https://github.com/user-attachments/assets/b0c7e62b-c359-4450-956e-4ba6df1db288)
![image](https://github.com/user-attachments/assets/1755943d-71e4-4522-8090-78867ecafb01)

**Reply by Jjmboni** [reply]
I was using net45-full.

**Comment by FreeAndNil** [comment, marked as answer]
Have you tested the last preview?
https://github.com/apache/logging-log4net/releases/tag/rc%2F3.0.0-preview.2

**Reply by Jjmboni** [reply]
The latest preview is working!  I installed v3.0.0-preview.2 and got no errors.  It works fine with MMA and Perfmon.dll.

Thanks!

**Reply by FreeAndNil** [reply]
Then it must a have been a dependency mismatch from the legacy frameworks.
Those were removed in 3.0.0.
